warl_proposal/warl_proposal.rst

```diff
@@ -0,0 +1,180 @@
+###################
+WARL field Proposal
```

Comment:
This proposal seems to describe all the implementation details of the WARL field, can you describe the use case of these information? It might be useful for simulation to generate or cover valid values. Do you expect it to be used in other places?

Reply:
You are right, the primary motivation was to define these so that compliance suite tests can be written in a more generic fashion.
 
Internally at IIT-Madras, we also plan to generate our hw csr modules using just the yaml and warl spec as inputs. 

Additionally, I hope for this YAML template to become the defacto standard of how any RISC-V implementation will be described in future. So a solid WARL spec which can account for future spec additions/modifications has to be in place. And i think this forms the basis of that effort.

Reply:
Sounds good, can you add me to the collaborator of this repo? Somehow I cannot resolve the conversation.

Reply:
added you as collaborator

Reply:
Thanks, somehow I cannot see the collaborator tag on my name, and I cannot resolve the conversation.

Reply:
I think you will have accept the invite from your mail-box maybe?

Reply:
It works now, thanks.